### PR TITLE
implement `dpnp.nanargmax`, `dpnp.nanargmin`, `dpnp.nanmax`, and `dpnp.nanmin`

### PR DESCRIPTION
**Pull request by vtavana**
In this PR, `dpnp.nanargmax`, `dpnp.nanargmin`, `dpnp.nanmax`, and `dpnp.nanmin` are implemented.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_statistics.py [hunk 1]
```diff
@@ -99,6 +101,30 @@ def test_max_min_error(func):
...
+    assert_dtype_allclose(dpnp_res, np_res)
+
+
+@pytest.mark.parametrize("func", ["nanmax", "nanmin"])
+def test_nanmax_nanmin_all_NaN(func):
+    a = numpy.arange(12, dtype=numpy.float32).reshape((2, 2, 3))
+    a[:, :, 2] = numpy.nan
+    ia = dpnp.array(a)
+
+    with warnings.catch_warnings():
```
Comment (antonwolfy) [inline tests/test_statistics.py:121]: that doesn't allow to check that dpnp and numpy raise the same warning, I'd suggest something like:
```python
def test_nanmax_nanmin_all_nan(recwarn, func):
    a = numpy.arange(12, dtype=numpy.float32).reshape((2, 2, 3))
    a[:, :, 2] = numpy.nan
    ia = dpnp.array(a)

    np_res = getattr(numpy, func)(a, axis=0)
    dpnp_res = getattr(dpnp, func)(ia, axis=0)
    assert_dtype_allclose(dpnp_res, np_res)

    assert len(recwarn) == 2
    assert all("All-NaN slice encountered" in str(r.message) for r in recwarn)
    assert all(r.category is RuntimeWarning for r in recwarn)
```
through `recwarn` pytest fixture

File: tests/third_party/cupy/sorting_tests/test_search.py [hunk 2]
```diff
@@ -366,7 +366,6 @@ def test_argwhere(self, xp, dtype):
 # return cupy.nonzero(self.array)
 
 
-@testing.gpu
 class TestNanArgMin(unittest.TestCase):
     @testing.for_all_dtypes(no_complex=True)
```
Comment (antonwolfy) [inline tests/third_party/cupy/sorting_tests/test_search.py:370]: for some below tests cupy uses `@testing.for_float_dtypes()`, I think we can align here

Comment (antonwolfy) [inline tests/third_party/cupy/sorting_tests/test_search.py:370]: also cupy doesn't have `accept_error=ValueError` in `numpy_cupy_allclose`, do we need that?

File: tests/third_party/cupy/sorting_tests/test_search.py [hunk 3]
```diff
@@ -366,7 +366,6 @@ def test_argwhere(self, xp, dtype):
 # return cupy.nonzero(self.array)
 
 
-@testing.gpu
 class TestNanArgMin(unittest.TestCase):
```
Comment (antonwolfy) [inline tests/third_party/cupy/sorting_tests/test_search.py:369]: to align with cupy
```suggestion
class TestNanArgMin:
```

File: tests/third_party/cupy/statistics_tests/test_order.py [hunk 4]
```diff
@@ -20,7 +20,6 @@ def for_all_interpolations(name="interpolation"):
     return testing.for_orders(_all_interpolations, name=name)
 
 
-@testing.gpu
 class TestOrder(unittest.TestCase):
```
Comment (antonwolfy) [inline tests/third_party/cupy/statistics_tests/test_order.py:23]: tests for `nanmin` and `nanmax` can be aligned with the latest cupy content

File: tests/test_mathematical.py [hunk 5]
```diff
@@ -718,66 +718,66 @@ def test_prod_zero_size(axis):
...
 
+    # out is a numpy array -> TypeError
     dpnp_res = numpy.empty_like(np_res)
     with pytest.raises(TypeError):
         getattr(dpnp, func)(ia, axis=0, out=dpnp_res)
 
+    # incorrect shape for out
     dpnp_res = dpnp.array(numpy.empty((2, 3)))
     with pytest.raises(ValueError):
         getattr(dpnp, func)(ia, axis=0, out=dpnp_res)
```
[the remark this answers is not in the document]
Comment (vtavana) [reply tests/test_mathematical.py:783]: I do not get this warning in my end. My python version  is 3.10.13.

[the remark this answers is not in the document]
Comment (antonwolfy) [reply tests/test_mathematical.py:785]: The same is reproduced in internal run, please check [log](https://cje-fm-owrp-prod04.devtools.intel.com/satg-dap-intelpython/job/intel-packages/job/dpnp/job/dev-linux-py3.10/job/test-wheel-conda/102/consoleFull), it was also python 3.10 used there.

**Review by antonwolfy — APPROVED** [review]
Thank you @vtavana , LGTM!

**Pull request merged**